py/server/deephaven/table_listener.py

```diff
@@ -355,20 +330,32 @@ def __init__(self, t: Table, listener: Union[Callable, TableListener], descripti
             listener (Union[Callable, TableListener]): listener for table changes
             description (str, optional): description for the UpdatePerformanceTracker to append to the listener's entry
                 description, default is None
+            dependencies (Union[Table, Sequence[Table]]): tables that must be satisfied before the listener's execution. A
+                refreshing table is considered to be satisfied if all updates to the table have been processed in the current
+                update graph cycle. A static table is always considered to be satisfied. If a specified table is refreshing,
+                it must belong to the same update graph as the table being listened to. Default is None.
+
+                Dependencies are used to ensure that the listener can safely access them during its execution, such as reading
+                the data from the tables or even performing table operations on them.
```

Comment:
> or even performing table operations on them

This is actually way less safe than you'd think.
New operations are OK, technically, but maybe inadvisable, because you might inherit memoized results, which you must depend on in order to access safely.